crates/ruff_linter/src/rules/pylint/rules/non_augmented_assignment.rs

```diff
@@ -135,21 +135,33 @@ pub(crate) fn non_augmented_assignment(checker: &mut Checker, assign: &ast::Stmt
 ///
 /// For example, given `x = x + 1`, the fix would be `x += 1`.
 fn augmented_assignment(
-    generator: Generator,
+    locator: &Locator,
     target: &Expr,
-    operator: Operator,
+    operator: AugmentedOperator,
     right_operand: &Expr,
     range: TextRange,
 ) -> Edit {
-    Edit::range_replacement(
-        generator.stmt(&ast::Stmt::AugAssign(StmtAugAssign {
-            range: TextRange::default(),
-            target: Box::new(target.clone()),
-            op: operator,
-            value: Box::new(right_operand.clone()),
-        })),
-        range,
-    )
+    let target_expr = locator.slice(target);
+    let right_operand_expr = locator.slice(right_operand);
+
+    let new_value_expr = if should_be_parenthesized_when_standalone(right_operand) {
+        format!("({right_operand_expr})")
+    } else {
+        right_operand_expr.to_string()
+    };
+    let new_content = format!("{target_expr} {operator} {new_value_expr}");
+
+    Edit::range_replacement(new_content, range)
+}
+
+/// Whether `expr` should be parenthesized when used on its own.
+///
+/// ```python
+/// a := 0            # (a := 0)
+/// a = b := 0        # a = (b := 0)
+/// ```
+const fn should_be_parenthesized_when_standalone(expr: &Expr) -> bool {
+    matches!(expr, Expr::Named(_))
```

Comment:
the `should_be_parenthesized_when_standalone()` function is so short, I feel like the code is easier to understand if the condition is just inlined:

```suggestion
    // expressions using the walrus operator (`:=`) must be parenthesized
    // when they're used as standalone expressions:
    let new_value_expr = if right_operand.is_named_expr() {
        format!("({right_operand_expr})")
    } else {
        right_operand_expr.to_string()
    };
    let new_content = format!("{target_expr} {operator} {new_value_expr}");

    Edit::range_replacement(new_content, range)
}
```

Reply:
This could also be done with just one allocation rather than two, but fixes aren't generally that performance-sensitive and it would make the code less readable.